share/eds/subscription.go

```diff
@@ -0,0 +1,40 @@
+package eds
+
+import (
+	"context"
+
+	pubsub "github.com/libp2p/go-libp2p-pubsub"
+)
+
+// Subscription is a wrapper over pubsub.Subscription that handles
+// receiving an EDS from other peers.
+type Subscription struct {
+	subscription *pubsub.Subscription
+}
+
+func newSubscription(t *pubsub.Topic) (*Subscription, error) {
+	subs, err := t.Subscribe()
+	if err != nil {
+		return nil, err
+	}
+
+	return &Subscription{subscription: subs}, nil
+}
+
+// Next blocks the callee until any new EDS notification(data hash) arrives.
+// Returns only notifications which successfully went through validation pipeline.
+func (subs *Subscription) Next(ctx context.Context) ([]byte, error) {
```

Comment:
Once #1428 is merged, we should use the DataHash type alias

Reply:
@vgonkivs, ok, it's merged, so we can use it now